internal/app/secretless/tcp_proxy_service_test.go

```diff
@@ -0,0 +1,126 @@
+package secretless
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin"
+)
+
+func TestNewTCPProxyService(t *testing.T) {
+	t.Run("empty constructor arguments result in errors", func(t *testing.T) {
+		_, err := NewTCPProxyService(
+			nil,
+			nil,
+			nil,
+		)
+		assert.Error(t, err)
+	})
+
+	t.Run("non-empty constructor arguments result in no error", func(t *testing.T) {
+		_, err := NewTCPProxyService(
+			new(connectorMock).connect,
+			new(credentialResolverMock).resolveCredentials,
+			new(listenerMock),
+		)
+		assert.NoError(t, err)
+	})
+
+	// TODO: propagates error from resolveCredentials
+	// TODO: resolveCredentials failure propagated
+
+	t.Run("zeroizes credentials from resolveCredentials", func(t *testing.T) {
+		connector := new(connectorMock)
+		credentialResolver := new(credentialResolverMock)
+		listener :=	new(listenerMock)
+
+		ps := tcpProxyService{
+			connector:          connector.connect,
+			resolveCredentials: credentialResolver.resolveCredentials,
+			listener:           listener,
+		}
+		creds := map[string][]byte{
+			"credName": []byte("credValue"),
+		}
+		credentialResolver.On("resolveCredentials").Return(
+			creds,
+			fmt.Errorf("couldn't resolve credentials"))
+
+		err := ps.handleConnection(nil)
+
+		assert.Error(t, err)
+		assert.Equal(
+			t,
+			string(creds["credName"]),
+			strings.Repeat("\x00", len([]byte("credValue"))),
+		)
+	})
+}
+
+func Test_tcpProxyService(t *testing.T) {
```

Comment:
Any reason this isn't `TestTCPProxyService`?

Reply:
It seems to be a Go convention for the test case to be Test_lowercaseSomething when it's an internal thing vs TestUpcaseSomething when it's a public thing.